src/main/java/com/example/accommodationbookingapp/service/user/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.example.accommodationbookingapp.service.user.impl;
+
+import com.example.accommodationbookingapp.dto.user.UserRegistrationRequestDto;
+import com.example.accommodationbookingapp.dto.user.UserResponseDto;
+import com.example.accommodationbookingapp.exception.RegistrationException;
+import com.example.accommodationbookingapp.mapper.UserMapper;
+import com.example.accommodationbookingapp.model.User;
+import com.example.accommodationbookingapp.repository.user.UserRepository;
+import com.example.accommodationbookingapp.service.user.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException(
+                    "Can't register a new user with email: "
+                            + requestDto.getEmail()
+            );
+        }
+        User user = userMapper.toModel(requestDto);
+        return userMapper.toDto(userRepository.save(user));
```

Comment:
i think u have to add this user.setPassword(passwordEncoder.encode(requestDto.getPassword()));

Reply:
I will implement this in next PR with all the security config

Reply:
> I will implement this in next PR with all the security config

ok